src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -492,50 +492,58 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             var hasErrors = localSymbol.ScopeBinder
                 .ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
-            BoundBlock block;
-            if (node.Body != null)
-            {
-                block = BindEmbeddedBlock(node.Body, diagnostics);
-            }
-            else if (node.ExpressionBody != null)
+            BoundBlock blockBody = node.Body != null ? BindEmbeddedBlock(node.Body, diagnostics) : null;
+            DiagnosticBag expressionBodyDiagnostics = null;
+            BoundBlock expressionBody = null;
+            if (node.ExpressionBody != null)
             {
-                block = BindExpressionBodyAsBlock(node.ExpressionBody, diagnostics);
+                expressionBodyDiagnostics = node.Body == null ? diagnostics : new DiagnosticBag();
+                expressionBody = BindExpressionBodyAsBlock(node.ExpressionBody, expressionBodyDiagnostics);
             }
-            else
+            else if (node.Body == null && node.ExpressionBody == null)
             {
-                block = null;
                 hasErrors = true;
                 diagnostics.Add(ErrorCode.ERR_LocalFunctionMissingBody, localSymbol.Locations[0], localSymbol);
             }
+            Debug.Assert(blockBody != null || expressionBody != null || hasErrors);
+            Debug.Assert((expressionBody == null) == (expressionBodyDiagnostics == null));
 
-            if (block != null)
+            if (blockBody != null || expressionBody != null)
             {
                 localSymbol.ComputeReturnType();
```

Comment:
> localSymbol.ComputeReturnType(); [](start = 15, length = 33)

It looks like this call is redundant, the ```GetDeclarationDiagnostics``` call is going to take care of this. I think we can remove this call and get rid of the ```if (blockBody != null || expressionBody != null)```.